### PR TITLE
[GEOT-6502] Add Geometry.class as context in FilterFunction_toWKT

### PR DESCRIPTION
**Pull request by taba90**
<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [X] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [X] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [X] PR for bug fixes and small new features are presented as a single commit
- [X] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [X] New unit tests have been added covering the changes
- [X] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by taba90** [comment]
Jira Ticket https://osgeo-org.atlassian.net/browse/GEOT-6502

**Comment by aaime** [comment]
@ianturton why? The test verifies that converters are being used, a simple evaluation of the Envelope literal would have failed the cast to Geometry, but now it's going through the GeometryConverterFactory. To me it seems a fair test, it would have failed without the change.

**Comment by aaime** [comment]
@ianturton I see the timings, maybe you haven't seen the test @taba90 pushed, your comment and his test are within 2 minutes from each other.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/appschema/filter/ToWKTTest.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package org.geotools.appschema.filter;
+
+public class ToWKTTest {}
```
Comment (ianturton) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/appschema/filter/ToWKTTest.java]: **This** test seems useless

Comment (aaime) [reply modules/extension/app-schema/app-schema/src/test/java/org/geotools/appschema/filter/ToWKTTest.java]: Doh! Yes, definitely useless :-D

Comment (taba90) [reply modules/extension/app-schema/app-schema/src/test/java/org/geotools/appschema/filter/ToWKTTest.java]: Sorry, indeed it wasn't supposed to be there, I've removed it.

**Pull request merged**